sonar-batch/src/main/java/org/sonar/batch/issue/tracking/InitialOpenIssuesSensor.java

```diff
@@ -58,26 +61,29 @@ public boolean shouldExecuteOnProject(Project project) {
 
   @Override
   public void analyse(Project project, SensorContext context) {
-    // Adding one second is a hack for resolving conflicts with concurrent user
-    // changes during issue persistence
-    final Date now = DateUtils.addSeconds(DateUtils.truncate(new Date(), Calendar.MILLISECOND), 1);
-
-    issueDao.selectNonClosedIssuesByModule(project.getId(), new ResultHandler() {
-      @Override
-      public void handleResult(ResultContext rc) {
-        IssueDto dto = (IssueDto) rc.getResultObject();
-        dto.setSelectedAt(now.getTime());
-        initialOpenIssuesStack.addIssue(dto);
-      }
-    });
+    ResourceDto module = resourceDao.getResource(ResourceQuery.create().setKey(project.getEffectiveKey()));
+    if (module != null) {
+      long moduleId = module.getId();
+      // Adding one second is a hack for resolving conflicts with concurrent user
+      // changes during issue persistence
+      final Date now = DateUtils.addSeconds(DateUtils.truncate(new Date(), Calendar.MILLISECOND), 1);
```

Comment:
No need to refactor this code as it will very soon be removed (batch will not read existing issue from the db but from a WS (already used in preview mode))